src/edit_mode/emacs.rs

```diff
@@ -62,3 +54,104 @@ impl Emacs {
         Emacs { keybindings }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use pretty_assertions::assert_eq;
+
+    #[test]
+    fn ctrl_l_leads_to_clear_screen_event() {
+        let mut emacs = Emacs::default();
+        let ctrl_l = Event::Key(KeyEvent {
+            modifiers: KeyModifiers::CONTROL,
+            code: KeyCode::Char('l'),
+        });
+        let result = emacs.parse_event(ctrl_l);
+
+        assert_eq!(result, ReedlineEvent::ClearScreen);
+    }
+
+    #[ignore = "Unsure what the desired behaviour is"]
+    #[test]
+    fn overriding_default_keybindings_works() {
+        let mut keybindings = default_emacs_keybindings();
+        keybindings.add_binding(
+            KeyModifiers::CONTROL,
+            KeyCode::Char('l'),
+            ReedlineEvent::HandleTab,
+        );
+
+        let mut emacs = Emacs::new(keybindings);
+        let ctrl_l = Event::Key(KeyEvent {
+            modifiers: KeyModifiers::CONTROL,
+            code: KeyCode::Char('l'),
+        });
+        let result = emacs.parse_event(ctrl_l);
+
+        assert_eq!(result, ReedlineEvent::HandleTab);
+    }
+
+    #[test]
+    fn inserting_character_works() {
+        let mut emacs = Emacs::default();
+        let l = Event::Key(KeyEvent {
+            modifiers: KeyModifiers::NONE,
+            code: KeyCode::Char('l'),
+        });
+        let result = emacs.parse_event(l);
+
+        assert_eq!(
+            result,
+            ReedlineEvent::Edit(vec![EditCommand::InsertChar('l')])
+        );
+    }
+
+    #[test]
+    fn inserting_capital_character_works() {
+        let mut emacs = Emacs::default();
+
+        let uppercase_l = Event::Key(KeyEvent {
+            modifiers: KeyModifiers::SHIFT,
+            code: KeyCode::Char('l'),
+        });
+        let result = emacs.parse_event(uppercase_l);
+
+        assert_eq!(
+            result,
+            ReedlineEvent::Edit(vec![EditCommand::InsertChar('L')])
+        );
+    }
+
+    #[ignore = "Unsure what the desired behaviour is"]
```

Comment:
either we can 
- return Option<ReedlineEvent>
- add some default (like NoOp) branch to ReedlineEvent
- return Result<ReedlineEvent> with a custom error type detailing what the problem is

I am not sure what the approach here should be. Thoughts?

Reply:
we should bake in a default keybinding. For things where there isn't a keybinding, I think we should return None.